driver/controller.go

```diff
@@ -191,6 +194,15 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	if snapshot != nil && vol != nil && volumeReq.SnapshotID != "" && int(volumeReq.SizeGigaBytes) > int(snapshot.SizeGigaBytes) && int(volumeReq.SizeGigaBytes) > 1 {
```

Comment:
Do we need to check `volume.SnapshotID` for emptiness? Shouldn't it be guaranteed to be non-empty when `snapshot` is non-nil?

Also, we don't seem to need the ID further down anyway AFAICT, so it looks to me as if we could ignore it in the if expression.